From 24ff5ee919eafe3a9a854f3e82077355317801c5 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Tue, 21 Sep 2021 12:30:28 +0800
Subject: [PATCH 25/30] Revert "drm/amdgpu: fix the doorbell missing when in
 CGPG issue for renoir."

This reverts commit 7525f2e4de0069983497a9d3eab1ca9813ae1b4b.
---
 drivers/gpu/drm/amd/amdgpu/gfx_v9_0.c | 21 +--------------------
 1 file changed, 1 insertion(+), 20 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/gfx_v9_0.c b/drivers/gpu/drm/amd/amdgpu/gfx_v9_0.c
index ac3a88197b2f..fb15e8b5af32 100644
--- a/drivers/gpu/drm/amd/amdgpu/gfx_v9_0.c
+++ b/drivers/gpu/drm/amd/amdgpu/gfx_v9_0.c
@@ -1271,16 +1271,6 @@ static bool is_raven_kicker(struct amdgpu_device *adev)
 		return false;
 }
 
-static bool check_if_enlarge_doorbell_range(struct amdgpu_device *adev)
-{
-	if ((adev->asic_type == CHIP_RENOIR) &&
-	    (adev->gfx.me_fw_version >= 0x000000a5) &&
-	    (adev->gfx.me_feature_version >= 52))
-		return true;
-	else
-		return false;
-}
-
 static void gfx_v9_0_check_if_need_gfxoff(struct amdgpu_device *adev)
 {
 	if (gfx_v9_0_should_disable_gfxoff(adev->pdev))
@@ -3629,16 +3619,7 @@ static int gfx_v9_0_kiq_init_register(struct amdgpu_ring *ring)
 	if (ring->use_doorbell) {
 		WREG32_SOC15(GC, 0, mmCP_MEC_DOORBELL_RANGE_LOWER,
 					(adev->doorbell_index.kiq * 2) << 2);
-		/* If GC has entered CGPG, ringing doorbell > first page
-		 * doesn't wakeup GC. Enlarge CP_MEC_DOORBELL_RANGE_UPPER to
-		 * workaround this issue. And this change has to align with firmware
-		 * update.
-		 */
-		if (check_if_enlarge_doorbell_range(adev))
-			WREG32_SOC15(GC, 0, mmCP_MEC_DOORBELL_RANGE_UPPER,
-					(adev->doorbell.size - 4));
-		else
-			WREG32_SOC15(GC, 0, mmCP_MEC_DOORBELL_RANGE_UPPER,
+		WREG32_SOC15(GC, 0, mmCP_MEC_DOORBELL_RANGE_UPPER,
 					(adev->doorbell_index.userqueue_end * 2) << 2);
 	}
 
-- 
2.27.0

